src/backend/access/transam/xlogreader.c

```diff
@@ -279,6 +279,14 @@ XLogReadRecord(XLogReaderState *state, char **errormsg)
 	bool		gotheader;
 	int			readOff;
 
+#define SKIP_INVALID_RECORD(rec_ptr)  do { \
```

Comment:
I wonder if rec_ptr should be aligned here.
Is there any warranty that it is maxaligned (8-bytes aligned) in all places where SKIP_INVALID_RECORD is called?

Reply:
there's sort of a defacto guarantee, with Page Header (both short and log) multiples of 8 and log records are maxaligned explicitly. Is it possible that log record in a corrupt  or alternatively formatted file are not aligned, so I'm a little on the fence on how I want to advance. but records from normally running system are always maxaligned